cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -472,6 +472,13 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return d_children[child_index];
   }
 
+  /**
+   * @brief Returns the number of child columns
+   *
+   * @return size_type The number of child columns
+   */
+  __device__ size_type num_child_columns() const noexcept { return _num_children; }
```

Comment:
Could we make this `__host__  __device__` ?

Reply:
done.

Reply:
Do we *need* to make it __host__ __device__? We should avoid doing so unless necessary.